src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java

```diff
@@ -650,57 +720,87 @@ private void refreshGrid(Grid<?> grid) {
     // A thread to load image data and update progress bar
     class ImageDataLoadingThread extends Thread {
         // Volatile because read in another thread in access()
-        volatile double img_count = 0.0;
+        volatile int img_count = 0;
+        volatile boolean interrupted = false;
 
         @Override
         public void run() {
 
-            img_count = 0.0;
+            img_count = 0;
 
             sampleImageMap.forEach( (imageId, ignoredImageName) -> {
-                HashMap<String, String> imageInformationMap = omeroClient.getImageInfo(selectedSample.getId(), imageId);
 
-                byte[] thumbnail = new byte[0];
-                String imageName = imageInformationMap.get("name");
-                String imageSize = imageInformationMap.get("size");
-                String imageTimePoints = imageInformationMap.get("tps");
-                String imageChannels = imageInformationMap.get("channels");
+                // continue if interrupted
+                if(interrupted){
+                    LOG.error("--> skipping loading iteration!");
+                    return;
+                }
 
                 try {
+                    HashMap<String, String> imageInformationMap = omeroClient.getImageInfo(selectedSample.getId(), imageId);
+
+                    String imageName = imageInformationMap.get("name");
+                    String imageSize = imageInformationMap.get("size");
+                    String imageTimePoints = imageInformationMap.get("tps");
+                    String imageChannels = imageInformationMap.get("channels");
+
                     ByteArrayInputStream thumbnailInputStream = omeroClient.getThumbnail(selectedSample.getId(), imageId);
-                    thumbnail = new byte[thumbnailInputStream.available()];
+                    byte[] thumbnail = new byte[thumbnailInputStream.available()];
                     // ignore integer and store in byte array
                     thumbnailInputStream.read(thumbnail);
                     thumbnailInputStream.close();
-                } catch (IOException ioException) {
-                    LOG.error("Could not retrieve thumbnail for image:" + imageId);
-                    LOG.debug(ioException);
-                }
 
-                ImageInfo imageInfo = new ImageInfo(imageId, imageName, thumbnail, imageSize, imageTimePoints, imageChannels);
-                imageInfos.add(imageInfo);
+                    ImageInfo imageInfo = new ImageInfo(imageId, imageName, thumbnail, imageSize, imageTimePoints, imageChannels);
+
+                    imageInfos.add(imageInfo);
+
+                    // update progress window
+                    img_count += 1;
+
+                } catch (Exception imgException) {
 
-                // update progress window
-                img_count += 1.0;
+                    interrupted = true;
```

Comment:
I'd keep the case of interruption seperate from a general loadup error that is supposed to be catched here
```suggestion
```

Reply:
I kinda used that to get the same cleanup as in the interruption exception, since we often get a connection error to OMERO when the thread is killed during data transfer. But you are correct in that it would be decoupled, it seems to be a a bit redundant also, I will comment it out and test again